### ISSUE TITLE
AppSpectorSDK 1.4.6 exception being thrown on app start

### ISSUE THREAD

**Issue by willk37** [appspector/ios-sdk#26]
There are times when launching our app that it crashes on start up. The issue seems to be related to AppSpectorSDK 1.4.6, I could not reproduce the issue on 1.4.5. This particular stack trace is from running a UI test that consistently crashes.

```
Invalid type in JSON write (NSError)
(
	0   CoreFoundation                      0x00007fff20406d44 __exceptionPreprocess + 242
	1   libobjc.A.dylib                     0x00007fff201a4a65 objc_exception_throw + 48
	2   Foundation                          0x00007fff208a9ca9 _writeJSONValue + 672
	3   Foundation                          0x00007fff208ae32e ___writeJSONObject_block_invoke + 350
	4   CoreFoundation                      0x00007fff2035e4e5 __NSDICTIONARY_IS_CALLING_OUT_TO_A_BLOCK__ + 7
	5   CoreFoundation                      0x00007fff20477c38 -[__NSDictionaryM enumerateKeysAndObjectsWithOptions:usingBlock:] + 255
	6   Foundation                          0x00007fff208ad8fb _writeJSONObject + 555
	7   Foundation                          0x00007fff208ae32e ___writeJSONObject_block_invoke + 350
	8   CoreFoundation                      0x00007fff2035e4e5 __NSDICTIONARY_IS_CALLING_OUT_TO_A_BLOCK__ + 7
	9   CoreFoundation                      0x00007fff20477c38 -[__NSDictionaryM enumerateKeysAndObjectsWithOptions:usingBlock:] + 255
	10  Foundation                          0x00007fff208ad8fb _writeJSONObject + 555
	11  Foundation                          0x00007fff208a99c8 -[_NSJSONWriter dataWithRootObject:options:] + 71
	12  Foundation                          0x00007fff208accf8 +[NSJSONSerialization dataWithJSONObject:options:error:] + 117
	13  AppSpectorSDK                       0x0000000112b22b2d -[ASSentryLogger logSentryEvent:] + 81
	14  AppSpectorSDK                       0x0000000112b22ac2 -[ASSentryLogger logEntry:] + 329
	15  AppSpectorSDK                       0x0000000112a8f4ab -[ASLogger logLevel:tag:extra:file:lineNumber:message:] + 544
	16  AppSpectorSDK                       0x0000000112a8f24e -[ASLogger logLevel:tag:extra:file:lineNumber:format:] + 267
	17  AppSpectorSDK                       0x0000000112aa69a3 -[ASHTTPCache cacheData:withKey:] + 313
	18  AppSpectorSDK                       0x0000000112aa6487 -[ASHTTPCache cacheResponse:body:withUUID:] + 130
	19  AppSpectorSDK                       0x0000000112ab260b -[ASHTTPRecorder recordLoadingFinishedWithRequestID:responseBody:] + 278
	20  AppSpectorSDK                       0x0000000112a7fcde __85+[ASNetworkObserver asyncCompletionWrapperForOriginalRequestID:mechanism:completion:]_block_invoke + 402
	21  CFNetwork                           0x00007fff23e39a92 CFNetwork + 31378
	22  CFNetwork                           0x00007fff23e55118 _CFHTTPMessageSetResponseProxyURL + 16297
	23  libdispatch.dylib                   0x00007fff201148e4 _dispatch_call_block_and_release + 12
	24  libdispatch.dylib                   0x00007fff20115b25 _dispatch_client_callout + 8
	25  libdispatch.dylib                   0x00007fff2011c0df _dispatch_lane_serial_drain + 753
	26  libdispatch.dylib                   0x00007fff2011ccf8 _dispatch_lane_invoke + 455
	27  libdispatch.dylib                   0x00007fff2012797b _dispatch_workloop_worker_thread + 779
	28  libsystem_pthread.dylib             0x00007fff6fb05fd0 _pthread_wqthread + 326
	29  libsystem_pthread.dylib             0x00007fff6fb04f57 start_wqthread + 15
)dyld4 config: DYLD_ROOT_PATH=/Applications/Xcode-13.3.0.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot DYLD_LIBRARY_PATH=/Users/<redacted>/Library/Developer/Xcode/DerivedData/BuzzFeed-hddfoxdzduycxlhjqvwmycwqlvux/Build/Products/Debug-iphonesimulator DYLD_INSERT_LIBRARIES=/Applications/Xcode-13.3.0.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/usr/lib/libMainThreadChecker.dylib DYLD_FRAMEWORK_PATH=/Users/<redacted>/Library/Developer/Xcode/DerivedData/BuzzFeed-hddfoxdzduycxlhjqvwmycwqlvux/Build/Products/Debug-iphonesimulator:/Users/<redacted>/Library/Developer/Xcode/DerivedData/BuzzFeed-hddfoxdzduycxlhjqvwmycwqlvux/Build/Products/Debug-iphonesimulator/PackageFrameworks
CoreSimulator 802.6 - Device: iPhone 8 (D68B2444-9D94-4E28-A20A-1F9802089E69) - Runtime: iOS 15.4 (19E240) - DeviceType: iPhone 8
```

**Comment by deszip** [comment]
Hi @willk37 , sorry for inconvenience, we'll take a look and get back with fix estimate.

**Comment by deszip** [comment]
Hi again @willk37, we investigated the issue and fix already gone to testing. Will be included in the next sdk release next week.

**Comment by deszip** [comment]
Issue fixed in SDK v 1.4.7. Feel free to reopen the issue if problem is still there.
